src/app/modules/form/admin-form/admin-form.service.ts

```diff
@@ -194,3 +197,62 @@ export const getMockSpcpLocals = (
       return {}
   }
 }
+
+/**
+ * Transfer form ownership from current owner to the new email.
+ * @param currentForm the form to transfer ownership for
+ * @param newOwnerEmail the email of the new owner to transfer to
+ *
+ * @return ok(updated form) if transfer is successful
+ * @return err(TransferOwnershipError) if new owner cannot be found in the database or new owner email is same as current owner
+ * @return err(DatabaseError) if any database errors like missing admin of current owner occurs
+ */
+export const transferFormOwnership = (
+  currentForm: IPopulatedForm,
+  newOwnerEmail: string,
+): ResultAsync<IPopulatedForm, TransferOwnershipError | DatabaseError> => {
+  const logMeta = {
+    action: 'transferFormOwnership',
+    newOwnerEmail,
+  }
+
+  // Step 1: Transfer form ownership.
+  return ResultAsync.fromPromise(
+    currentForm.transferOwner(newOwnerEmail),
+    (error) => {
+      logger.error({
+        message: 'Error occurred whilst transferring form ownership',
+        meta: logMeta,
+        error,
+      })
+
+      // Special case, when document instance method already returns
+      // ApplicationError, directly return the error.
+      if (
+        error instanceof TransferOwnershipError ||
+        error instanceof DatabaseError
+      ) {
+        return error
+      }
+
+      return new DatabaseError(getMongoErrorMessage(error))
```

Comment:
i'm not comfortable with returning a `DatabaseError` by default, as this means that the service implementation is aware of the internals of the domain layer. do we have a more generic error type, or should we simply forward whichever error is returned from the `transferOwner` function?

Reply:
DatabaseError is the generic type for database errors though. And seems to be appropriate seeing that the service is calling a database function, the only possible errors would be a database error.